tests/OSPSuite.Core.IntegrationTests/DiffBuilders/ContainerDiffBuilderSpecs.cs

```diff
@@ -1,99 +1,46 @@
 ﻿using System.Linq;
+using FakeItEasy;
+using NUnit.Framework;
 using OSPSuite.BDDHelper;
 using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Core.Comparison;
 using OSPSuite.Core.Domain;
 
 namespace OSPSuite.Core.DiffBuilders
 {
-   public class When_comparing_two_container_having_different_base_properties_and_container_properties : concern_for_ObjectComparer
+   public class concern_for_ContainerDiffBuilder : ContextSpecification<ContainerDiffBuilder>
    {
-      protected override void Context()
-      {
-         base.Context();
-         _comparerSettings.OnlyComputingRelevant = false;
-         _object1 = new Container {ContainerType = ContainerType.Formulation, Name = "O1"};
-         _object2 = new Container {ContainerType = ContainerType.Event, Name = "O2"};
-      }
-
-      [Observation]
-      public void should_return_a_report_containing_the_name_and_the_container_type_difference()
-      {
-         _report.Count().ShouldBeEqualTo(2);
-      }
-   }
+      protected IComparison<IContainer> _comparison;
+      protected DiffReport _report;
 
-   public class When_comparing_two_container_having_the_same_number_of_children_but_with_differences : concern_for_ObjectComparer
-   {
       protected override void Context()
       {
-         base.Context();
-
-         var container1 = new Container {Name = "O1"};
-         container1.Add(new Parameter {BuildMode = ParameterBuildMode.Global, Name = "P1"});
-
-         var container2 = new Container {Name = "O2"};
-         container2.Add(new Parameter {BuildMode = ParameterBuildMode.Local, Name = "P1"});
-
-         _object1 = container1;
-         _object2 = container2;
-      }
+         var entityDiffBuilder = A.Fake<EntityDiffBuilder>();
```

Comment:
You are moving from Integration tests actually testing the real behavior to using fake. I don't understand the reasoning. In fact, why are you not using concern_for_ObjectComparer ? This does everything for you. You just have to set _object1 and object2 in all your subtests and everything is done for you. RIght now, you are reinvitening the wheel

Reply:
it does not in fact, since the change I did is on the ContainerDiffBuilder Class, which was not being tested at all.
The tests (even when the class was called "ContainerDiffBuilderSpecs" is testing the "ObjectComparer" class, not the ContainerDiffBuilder.
And since my changes are on the ContainerDiffBuilder class, I created this tests and changed the previous class name as well.